### PR TITLE
Fix branch calculator

### PR DESCRIPTION
**Pull request by Evhenii99**
This PR added 2 calculators. My firs program

### CODE REVIEW

File: Another projekts/calculator/main_calculator_1.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
 new = True
 
+
+
+
+
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py]: Please delete those 4 extra lines.

File: Another projekts/calculator/main_calculator_2.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 print("Our Magical Calculator")
 print("Type 'quit' to exit\n")
 
+
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_2.py]: Please delete the extra line.

Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_2.py]: Can you **replace** duplicated code 
```
            first_number = int(input("Enter first number: "))
            second_number = int(input("Enter second number: "))
            result = first_number + second_number
            print(f"Answer is {result}")
            user_input = input("Do you want to continue? Enter Yes or No: ")
            user_input = user_input.capitalize()
            if user_input == 'Yes':
                new = True
            else:
                break
```
     
with some `operate(param1 param2)` function?

Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_2.py]: Could you apply this:
```
            if user_input == 'Yes':
                new = True
            else:
                break
```

To the calculator 2.

Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_2.py]: I will do it tomorrow!

File: Another projekts/calculator/main_calculator_1.py (hunk 1 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > Будь ласка, видаліть ці 4 зайві рядки.

Done

File: Another projekts/calculator/main_calculator_2.py (hunk 2 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_2.py]: > 

Done

Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_2.py]: > Can you **replace** duplicated code
> 
> ```
>             first_number = int(input("Enter first number: "))
>             second_number = int(input("Enter second number: "))
>             result = first_number + second_number
>             print(f"Answer is {result}")
>             user_input = input("Do you want to continue? Enter Yes or No: ")
>             user_input = user_input.capitalize()
>             if user_input == 'Yes':
>                 new = True
>             else:
>                 break
> ```
> 
> with some `operate(param1 param2)` function?

Done

Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_2.py]: > Could you apply this:
> 
> ```
>             if user_input == 'Yes':
>                 new = True
>             else:
>                 break
> ```
> 
> To the calculator 2.

This was used in calculator_1

File: Another projekts/calculator/main_calculator_1.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    match what_todo:
+        case '1':
+            mathematical_operation()
+        case '2':
+            mathematical_operation()
+        case '3':
+            mathematical_operation()
+        case '4':
+            mathematical_operation()
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py]: You can add a parameter to `mathematical_operation()` like this:
```
def mathematical_operation(operation):
```
This is a good code style. Better than using the global variables. This makes your function portable to the other code.
And you can avoid `match-case` duplication as well.
Can you see the duplication? 
`case '1':` is equal to `if what_todo == '1':`
Basically, you need first check if there is no `_` matching and then execute `mathematical_operation(what_todo)` in a `while run:` loop.

File: Another projekts/calculator/main_calculator_1.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        case _:
+            print("Your value in not 1-4")
+
+    user_input = input("Do you want to continue? Enter Yes or No: ")
+    user_input = user_input.capitalize()
+    if user_input == 'Yes':
+        run = True
+    else:
+        break
+print("Thank for using this program, have a nice day!")
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py]: Please don't forget the last empty line.

File: Another projekts/calculator/main_calculator_1.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+run = True
+
+while run:
+    what_todo = input("What do you want to do?\n1 for +\n2 for -\n3 for *\n4 for /\n: ")
+
+    def mathematical_operation():
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py]: This is an independent function. Is there any Python standard that requires function creation in the while loop?

File: Other projekts/calculator/main_calculator.py [hunk 6]
```diff
@@ -1,54 +0,0 @@
-new = True
```
Comment (kozak-yakym) [inline Other projekts/calculator/main_calculator.py:1]: Why did you remove this code?

File: Another projekts/calculator/main_calculator_1.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        second_number = int(input("Enter second number: "))
+        if what_todo == '1':
+            result = first_number + second_number
+        elif what_todo == '2':
+            result = first_number - second_number
+        elif what_todo == '3':
+            result = first_number * second_number
+        elif what_todo == '4':
+            result = first_number / second_number
+        print(f"Answer is {result}")
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py]: Isn't it better to output "**The** answer is ..."?

File: Another projekts/calculator/main_calculator_1.py (hunk 4 above)
Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_1.py]: image
 
Beware always those signs in your PR.

File: Another projekts/calculator/main_calculator_1.py (hunk 7 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > Isn't it better to output "**The** answer is ..."?
Yes, you are right i am going to do that

File: Another projekts/calculator/main_calculator_1.py (hunk 4 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > Please don't forget the last empty line.
Thanks, i fixed that

File: Other projekts/calculator/main_calculator.py (hunk 6 above)
Comment (Evhenii99) [reply Other projekts/calculator/main_calculator.py:1]: > Why did you remove this code?

I added variable run, it's better to understand as for me

File: Another projekts/calculator/main_calculator_1.py (hunk 5 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > This is an independent function. Is there any Python standard that requires function creation in the while loop?

This function is independent, I can take it out of the while loop.
For example, write it before the cycle

File: Another projekts/calculator/main_calculator_1.py (hunk 3 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > You can add a parameter to `mathematical_operation()` like this:
> 
> ```
> def mathematical_operation(operation):
> ```
> 
> This is a good code style. Better than using the global variables. This makes your function portable to the other code. And you can avoid `match-case` duplication as well. Can you see the duplication? `case '1':` is equal to `if what_todo == '1':` Basically, you need first check if there is no `_` matching and then execute `mathematical_operation(what_todo)` in a `while run:` loop.

I deleted match case and add a parameter in ```mathematical_operation()```
But i don't understand everything and written code some different.
Now you can check

File: Another projekts/calculator/main_calculator_1.py (hunk 7 above)
Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_1.py]: > Isn't it better to output "**The** answer is ..."?

Have you done it yet?

Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > > Isn't it better to output "**The** answer is ..."?
> 
> Have you done it yet?

Yes, i did it

File: Another projekts/calculator/main_calculator_1.py (hunk 3 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: > You can add a parameter to `mathematical_operation()` like this:
> 
> ```
> def mathematical_operation(operation):
> ```
> 
> This is a good code style. Better than using the global variables. This makes your function portable to the other code. And you can avoid `match-case` duplication as well. Can you see the duplication? `case '1':` is equal to `if what_todo == '1':` Basically, you need first check if there is no `_` matching and then execute `mathematical_operation(what_todo)` in a `while run:` loop.

Test answer

Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: test answer 1

File: Another projekts/calculator/main_calculator_1.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
+run = True
+
+while run:
+    first_number = int(input("Enter first number: "))
+    second_number = int(input("Enter second number: "))
+    what_todo = input("What do you want to do?\n1 for +\n2 for -\n3 for *\n4 for /\n: ")
+
+    def mathematical_operation(choise):
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py]: Just move this out of the while: loop.
As far as I know, it's a bad coding style that you have here (to define functions inside others).
Please brush up on your knowledge of the functions and main context.

Defining a function inside a while loop in Python is generally not considered good practice, and it can cause issues in certain situations. Here are a few reasons why:

- Performance: Defining a function inside a loop means that the function will be redefined every time the loop iterates. This can lead to unnecessary overhead and slow down the execution of the code.
- Code clarity: Defining a function inside a loop can make the code less readable and harder to understand, especially if the loop is long or complex.
- Scope: If the function is defined inside a loop, it will only be accessible within that loop. This can cause issues if you need to use the function outside of the loop or in other parts of your code.

In general, it is better to define functions outside of loops or in a separate module or file, where they can be easily reused and shared across multiple parts of your code.

File: Another projekts/calculator/main_calculator_1.py (hunk 3 above)
Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_1.py]: This is the way to go except for the function definition.
I hope you can refactor more this code.

File: Another projekts/calculator/main_calculator_1.py (hunk 5 above)
Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_1.py]: Please do.

File: Other projekts/calculator/main_calculator.py (hunk 6 above)
Comment (kozak-yakym) [reply Other projekts/calculator/main_calculator.py:1]: Ok. Thanks for the clarification.

File: Another projekts/calculator/main_calculator_1.py (hunk 7 above)
Comment (kozak-yakym) [reply Another projekts/calculator/main_calculator_1.py]: Thanks for the feedback.

File: Another projekts/calculator/main_calculator_1.py (hunk 8 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: I exported the function to another file and imported it into the working program.
First_number and second_number i added in inside function in another file which i import to my main file.
I can move out the what_todo, but then user cant choose what to do in next iteration always will be completed the operation whose he choice in first time

File: Another projekts/calculator/main_calculator_1.py (hunk 5 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py]: Done

File: Another projekts/calculator/functions.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+        result = first_number - second_number
+        print(f"The answer is {result}")
+    elif choise == '3':
+        result = first_number * second_number
+        print(f"The answer is {result}")
+    elif choise == '4':
+        result = first_number / second_number
+        print(f"The answer is {result}")
+    else:
+        print(f"Your value in not 1-4")
```
Comment (kozak-yakym) [inline Another projekts/calculator/functions.py]: Why does it happen again? (No Last empty line)
Please look at the File changes tab and re-check carefully all your changes in code before you ask for the code review.

File: Python_new_2023/Section 8/test1.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+run = True
```
Comment (kozak-yakym) [inline Python_new_2023/Section 8/test1.py:1]: Please explain why you need to add this file. 
It would be great if you edit the PR description accordingly.

File: Another projekts/calculator/main_calculator_1.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
+from functions import mathematical_operation
```
Comment (kozak-yakym) [inline Another projekts/calculator/main_calculator_1.py:1]: That's a great solution for your studying in my opinion but do not overdo this in the other projects. This is can be a false friend when you have a really simple program and not planning to use this module in other projects. In this case, such complications can make worse code clarity.

**Review by kozak-yakym — APPROVED** [review]
In common this is good to go. Great Job!
But next time try to create PR only for the feature. This means the other changes such as Folders or Files clean-up (re-organizing), other than feature-related code clean-up or refactoring make a lot of headaches for the reviewer. It's better to do it in a separate PR.
And please mention all the changes (in common. just topics) and concepts in the description.
I Approve this but fix please the PR's description.

File: Another projekts/calculator/main_calculator_1.py (hunk 11 above)
Comment (Evhenii99) [reply Another projekts/calculator/main_calculator_1.py:1]: I will take that into consideration

File: Python_new_2023/Section 8/test1.py (hunk 10 above)
Comment (Evhenii99) [reply Python_new_2023/Section 8/test1.py:1]: In my case it is not necessary, I can use - While True
But at the time of creating the code, it seemed to me the right solution

File: Another projekts/calculator/functions.py (hunk 9 above)
Comment (Evhenii99) [reply Another projekts/calculator/functions.py]: Missed it.
I wouldn't be able to compare it to the previous file because that file was created last time
